docs: add documentation about `noPrimaryKey` (#109)
sequelize/website: docs/core-concepts/model-basics.md

});
```

## Prevent creating a default PK attribute

> By default, Sequelize automatically adds the primary key attribute `id` to every model when there is no primary key has been defined manually. To prevent this you can use the option `noPrimaryKey` when creating the instance and set it to true.

Reviewer: I left an extra `there is` during my previous review 😅 

Extra suggestion: reworded the second sentence because the example that follows doesn't show a creation of a Sequelize instance anymore but the creation of a model :)

```suggestion
By default, Sequelize automatically adds the primary key attribute `id` to every model when no primary key has been defined manually. To prevent this you can set the `noPrimaryKey` option to true when defining the model.
```

```js
const User = sequelize.define('User', {
  name: DataTypes.STRING,
}, {
  noPrimaryKey: true,
});
```

> In case you want to disable primary keys on every models.

Reviewer: What do you think of this rewording? I'm proposing it because I'm afraid someone may think it means all primary keys will be disabled, even the ones added manually

```suggestion
In case you want to prevent the addition of the default foreign key for every model:
```

Author: > 

I think it's nice!

Author: It's my mistake to not aware about it

> In case you want to prevent the addition of the default foreign key for every model:

Reviewer: Damn it, this is me again

```suggestion
In case you want to prevent the addition of the default primary key for every model:
```

});
```

## Prevent creating a default PK attribute

> By default, Sequelize automatically adds the primary key field `id` to every model when there is no primary key field defined and would throw an error when there is an `id` field but not a primary key field, to prevent this you can use the option `noPrimaryKey` and set it to true.

Reviewer 2: ```suggestion
By default, Sequelize automatically adds the primary key field `id` to every model when there is no primary key field defined. When there is an `id` field but not a primary key field Sequelize will throw an error. To prevent this you can use the option `noPrimaryKey` when creating the instance and set it to true.
```

Just a suggestion to make it more readable. Otherwise it is quite a long sentence

> By default, Sequelize automatically adds the primary key field `id` to every model when there is no primary key field defined. When there is an `id` field but not a primary key field Sequelize will throw an error. To prevent this you can use the option `noPrimaryKey` when creating the instance and set it to true.

Reviewer: A small change because I would prefer to avoid using the word `field` and instead always refer to it as either `column` (when talking about sql) or `attribute` (when talking about js)

I would also propose to remove the part about sequelize throwing an error if `id` is added manually, but its `primaryKey` option is not configured for a few reasons:

- it's unrelated to `noPrimaryKey` (it works as a "hey, did you mean that?")
- the error it throws is clear enough
- it's not necessary for the user to know about it until they encounter the error, at which point the error will explain what is going on. They're never going to actively seek this information.

```suggestion
By default, Sequelize automatically adds the primary key attribute `id` to every model when there is no primary key has been defined manually. To prevent this you can use the option `noPrimaryKey` when creating the instance and set it to true.
```

Author: > A small change because I would prefer to avoid using the word `field` and instead always refer to it as either `column` (when talking about sql) or `attribute` (when talking about js)
> 
> I would also propose to remove the part about sequelize throwing an error if `id` is added manually, but its `primaryKey` option is not configured for a few reasons:
> 
> * it's unrelated to `noPrimaryKey` (it works as a "hey, did you mean that?")
> * the error it throws is clear enough
> * it's not necessary for the user to know about it until they encounter the error, at which point the error will explain what is going on. They're never going to actively seek this information.

Interesting explanation 😄 , I will continue with resolving the other tomorrow.

```js

> const sequelize = Sequelize.createSequelizeInstance({

Reviewer: `Sequelize.createSequelizeInstance` does not exist, it's a helper we use in our tests.

The way to create a sequelize instance is just `new Sequelize`

Reviewer 2: Woops, I missed that. Good one!

  define: {
    noPrimaryKey: true,
  },
});

const User = sequelize.define('User', {
  name: {
    type: DataTypes.STRING
  }
});
```

> In case you only want to disable primary keys on a single model, you can define the field with `primaryKey: false`.

Reviewer: That's not true. Doing that will still add a primary key field called 'id'. It's only going to make `name` not a primary key, which would already be the case without the option.

This is how you disable adding the primary key using local options:

```js
const User = sequelize.define('User', {
  name: DataTypes.STRING,
}, {
  noPrimaryKey: true,
});
```

I would also suggest switching the order of explanations to start with the local options first, then the global ones (as we always do throughout the documentation)

> In case you do not want to use `noPrimaryKey` and do not want to caught by an error you can define `primaryKey` on the model fields and set it to false.

Reviewer 2: Maybe rephrase this so it's more clear that this will then happen only on a single model.

Something like: 'In case you only want to disable primary keys on a single model, you can define the field with `primaryKey: false`'

Author: Thanks
